### PR TITLE
Remove unneeded dplyr methods

### LINKED ISSUE #177: Overlapping methods in connectapi break dbplyr functionality
Something unexpected is happening with connectapi's custom `tbl_lazy` methods for `op_vars`, `dim` and `dimnames` generics (some of which are probably not necessary with the latest version of dbplyr).

I have some code that relies on `colnames()`, where it's expected that `colnames()` returns the column names of the final table (if we take out `x`, then it shouldn't include `x`).

``` r
library(dplyr)
library(dbplyr)

mf <- memdb_frame(g = c(1:3), x = 3:1, y = 3:1 + 3)
mf
#> # Source:   table<dbplyr_001> [?? x 3]
#> # Database: sqlite 3.37.0 [:memory:]
#>       g     x     y
#>   <int> <int> <dbl>
#> 1     1     3     6
#> 2     2     2     5
#> 3     3     1     4
mf %>% colnames()
#> [1] "g" "x" "y"
mf %>% select(-x) %>% colnames()
#> [1] "g" "y"
```

But when I load `connectapi`, `colnames()` grabs the colnames from the wrong place (from the `x` of the select operation object), and gives an incorrect answer.

```r
library(connectapi)
#> Registered S3 methods overwritten by 'connectapi':
#>   method            from  
#>   dim.tbl_lazy      dbplyr
#>   dimnames.tbl_lazy dbplyr
mf %>% colnames()
#> [1] "g" "x" "y"
mf %>% select(-x) %>% colnames()
#> [1] "g" "x" "y"
```

Similarly, joining is weird:

``` r
library(dplyr)
library(dbplyr)

mf1 <- memdb_frame(g = c(1:3), x = 3:1)
mf2 <- memdb_frame(g = c(1:3), y = 3:1 + 3)

left_join(mf1, mf2)
#> Joining, by = "g"
#> # Source:   lazy query [?? x 3]
#> # Database: sqlite 3.37.0 [:memory:]
#>       g     x     y
#>   <int> <int> <dbl>
#> 1     1     3     6
#> 2     2     2     5
#> 3     3     1     4

library(connectapi)
#> Registered S3 methods overwritten by 'connectapi':
#>   method            from  
#>   dim.tbl_lazy      dbplyr
#>   dimnames.tbl_lazy dbplyr
left_join(mf1, mf2)
#> Joining, by = "g"
#> Error in UseMethod("op_vars"): no applicable method for 'op_vars' applied to an object of class "c('op_join', 'op_double', 'op')"
```

I think both cases trace back to the `op_vars` methods in connectapi, and from [looking at dplyr's source](https://github.com/tidyverse/dbplyr/blob/70f408517a20514aa68c87d356e2c7f272d8f562/R/tbl-lazy.R#L108), I'm not sure if the overlapping methods are even necessary anymore.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #177. Closes #208.

---

Overall, my belief is that this is an innocuous change because the dplyr methods should be called automatically, and it's not clear to me why these were ever needed. That said, since I wasn't sure when the integration tests were automatically run, I also tested by hand by verifying that the output from these calls looked ok:

```R
conn <- connect("https://colorado.posit.co/rsc/")

tbl_connect(conn, "users")
tbl_connect(conn, "usage_static")
tbl_connect(conn, "content")
tbl_connect(conn, "groups")
```

Once this is merged, I think it would be useful to get a patch release to CRAN.

### CODE REVIEW

**Comment by hadley** [comment]
Looks like the integration tests are run on GHA, and do suggest a problem:

``` r
library(connectapi)
conn <- connect("https://colorado.posit.co/rsc/")
#> Defining Connect with server: https://colorado.posit.co/rsc/
#> Warning: You are using a newer version of Posit Connect (2023.12.0) than was tested (2022.09.0). Most APIs should function as expected.
#> This warning is displayed once per session.

users <- tbl_connect(conn, "users")
users
#> Error in if (is.na(rows) || rows > get_pillar_option_print_max()) {: missing value where TRUE/FALSE needed
nrow(users)
#> NULL
ncol(users)
#> NULL

db <- dbplyr::memdb_frame(x = 1:10)
nrow(db)
#> [1] NA
ncol(db)
#> [1] 1
```

<sup>Created on 2024-02-14 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

Let me dig into this

**Comment by hadley** [comment]
Ok, I think the main confusion here was essentially creating a new generic with the same name as the dbplyr generic (`op_vars`). Once I figured out that was it's own thing, the other pieces came together quickly.

**Review by colearendt — APPROVED** [review]
Thanks for digging into this and resolving those issues!

**Comment by hadley** [comment]
I also realised that there's an implicit dbplyr dependency here, since that's the package that provides most of the `tbl_lazy` methods; I've made that more explicit.

File: R/lazy.R [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #'
 #' @export
 tbl_connect <- function(src, from = c("users", "groups", "content", "usage_shiny", "usage_static", "audit_logs"), ...) {
+  rlang::check_required("dbplyr")
```
Comment (olivroy) [inline R/lazy.R:23]: @hadley this should be `rlang::check_installed()` ;)

**Pull request merged**